src/mapping/query/function/mod.rs

```diff
@@ -228,16 +233,18 @@ impl Argument {
 }
 
 impl Function for Argument {
-    fn execute(&self, ctx: &Event) -> Result<Value> {
+    fn execute(&self, ctx: &Event) -> Result<QueryValue> {
         let value = self.resolver.execute(ctx)?;
 
         // Ask the parameter if it accepts the given value.
-        if !(self.parameter.accepts)(&value) {
-            return Err(format!(
-                "invalid argument type '{}' for parameter '{}'",
-                value.kind(),
-                self.parameter.keyword
-            ));
+        if let QueryValue::Value(ref value) = value {
+            if !(self.parameter.accepts)(&value) {
+                return Err(format!(
+                    "invalid argument type '{}' for parameter '{}'",
+                    value.kind(),
+                    self.parameter.keyword
+                ));
+            }
```

Comment:
Would be good to add a `TODO` here w.r.t. also checking non `Value` kinds (e.g. `Regex`, etc).

Perhaps make the `if let` into a `match`, something like:

```rust
match &value {
	QueryValue::Value(value) => { ... },
	QueryValue::Regex => { /* TODO: ... */ },
}
```